strings/base_iterator.h

```diff
@@ -0,0 +1,183 @@
+
+namespace winrt::impl
+{
+    template <typename T>
+    struct fast_iterator
+    {
+        using iterator_concept = std::random_access_iterator_tag;
+        using iterator_category = std::input_iterator_tag;
+        using value_type = decltype(std::declval<T>().GetAt(0));
+        using difference_type = ptrdiff_t;
+        using reference = value_type;
+
+        fast_iterator() noexcept : m_collection(nullptr), m_index(0) {}
```

Comment:
Could this use member initializer and then default the constructor?

Reply:
Indeed, done